backends/verilog/verilog_backend.cc

```diff
@@ -1918,9 +1955,13 @@ void dump_sync_print(std::ostream &f, std::string indent, const RTLIL::SigSpec &
 	for (auto cell : cells) {
 		f << stringf("%s" "  if (", indent.c_str());
 		dump_sigspec(f, cell->getPort(ID::EN));
-		f << stringf(")\n");
+		f << stringf(") begin\n");
 
 		dump_cell_expr_print(f, indent + "    ", cell);
```

Comment:
For `$check` I think we want `if (!A)` in front of the print statement

Reply:
Oh yeah, good catch!

Reply:
`if (!A)` is good for `assert`, what about the other flavors?

Reply:
I think they're all good with `if (!A)`

Reply:
Wait, no. `A` is emitted as a part of the `assert`/etc itself. It doesn't need to be added the second time. (Which is the whole point of having it.)

Reply:
I am confused. It is part of the `assert`, but e.g. if we roundtrip Verilog through Yosys we don't want to print `assert property ... failed` each cycle the assertion is enabled.

Reply:
Ohhh, you meant for the `$print` part! Yes, absolutely.

Reply:
Check out my latest update, where I think I've handled it.